File: src/main/java/ash/java/graphql/data/TvDaoImpl.java
package ash.java.graphql.data;

import ash.java.graphql.types.tvseason.TvSeasonType;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.mashape.unirest.http.HttpResponse;
import com.mashape.unirest.http.JsonNode;
import org.springframework.stereotype.Service;

import java.lang.reflect.Type;

@Service
public class TvDaoImpl implements TvDao {

    private Gson gson = new Gson();
    private Type tvSeasonType = new TypeToken<TvSeasonType>(){}.getType();

    @Override
    public TvSeasonType getTvSeason(int tvShowId, int seasonNumber) {
        HttpResponse<JsonNode> response = TmdbHttpUtils.sendRequest(TmdbUrls.TmdbTwoArgUrl.TV_SEASON_URL,
                Integer.toString(tvShowId),
                Integer.toString(seasonNumber));

        String responseJson = response.getBody().getObject().toString();

        return gson.fromJson(responseJson, tvSeasonType);
    }
}

File: src/main/kotlin/ash/kotlin/graphql/data/TvDaoImpl.kt
package ash.kotlin.graphql.data

import ash.kotlin.graphql.types.tvseason.TvSeasonType
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.mashape.unirest.http.HttpResponse
import com.mashape.unirest.http.JsonNode

class TvDaoImpl : TvDao
{
  override fun getTvSeason(tvShowId: Int, seasonNumber: Int): TvSeasonType
  {
    val response: HttpResponse<JsonNode> = TmdbRequest().sendTwoArgRequest(TmdbTwoArgUrl.TV_SEASON_URL,
        tvShowId.toString(),
        seasonNumber.toString())
    val responseJson = response.body.`object`.toString()

    return Gson().fromJson(responseJson, object : TypeToken<TvSeasonType>()
    {}.type)
  }
}